### PR TITLE
Nova Placement API

### PR DESCRIPTION
**Pull request by jyotty**

### CODE REVIEW

File: roles/nova-common/defaults/main.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ nova:
       config_file: libvirt.conf
       defaults: /etc/sysconfig/libvirtd
       group: libvirt
+  nova_lib_dir: "/opt/openstack/current/nova"
```
Comment (nirajdp76) [inline roles/nova-common/defaults/main.yml]: how are we handling path for rhosp?

Comment (nirajdp76) [inline roles/nova-common/defaults/main.yml]: Also we need to ensure following for distro path
- ensure openstack-nova-placement rpm is being installed  (will have to check for exact rpm name). This is done by adding rpm to project_packages list in nova-common/defaults (for distro)
- add nova placement service defn to https://github.com/blueboxgroup/ursula/blob/master/roles/openstack-meta/defaults/main.yml#L113  . Follow existing pattern to stay consistent.
- ensure systemd override is getting written out - https://github.com/blueboxgroup/ursula/blob/master/roles/nova-control/tasks/main.yml#L23 - Follow same pattern, we write override systemd defn in /etc/systemd.

File: roles/nova-common/tasks/main.yml [hunk 2]
```diff
@@ -74,6 +74,26 @@
...
+      - reload apache
+
+  - name: enable placement api apache site
+    apache2_site: name=nova_placement_api state=present
+    notify:
+      - reload apache
+  when: ursula_os == 'ubuntu'
+
+- name: placement api apache config (rhel)
+  template: src=etc/httpd/conf.d/nova_placement_api.conf
```
Comment (nirajdp76) [inline roles/nova-common/tasks/main.yml]: Where is this template file?

File: roles/nova-common/templates/etc/apache2/sites-available/nova_placement_api.conf [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Listen 9775
+WSGIPythonHome {{ nova.nova_lib_dir }}
+WSGIPythonPath /etc/nova
+
+<VirtualHost *:8080>
```
Comment (nirajdp76) [inline roles/nova-common/templates/etc/apache2/sites-available/nova_placement_api.conf]: 8080?

Comment (nirajdp76) [inline roles/nova-common/templates/etc/apache2/sites-available/nova_placement_api.conf]: should we be using uswgi?  you might want to check with Zack.

Comment (jyotty) [inline roles/nova-common/templates/etc/apache2/sites-available/nova_placement_api.conf]: It ships with nova as a uswgi script. https://github.com/openstack/nova/blob/stable/newton/setup.cfg#L72-L73
https://docs.openstack.org/developer/nova/placement.html#steps

File: roles/nova-common/tasks/main.yml [hunk 4]
```diff
@@ -74,6 +74,60 @@
     mode: 0640
   when: inventory_hostname in groups['controller']
 
+- name: set uwsgi path (source install)
```
Comment (nirajdp76) [inline roles/nova-common/tasks/main.yml]: Is nova-placement-api service to be install on controllers and computes? if is just controllers than this all probably should move to nova-control role.

**Comment by nirajdp76** [comment]
retest

**Comment by jyotty** [comment]
@nirajdp76 has added changes and opened #2819, closing this.

**Pull request closed without merging**